src/simplify_algebra.cpp

```diff
@@ -78,6 +77,162 @@ struct find_mul_conv
     }
 };
 
+struct find_resize
+{
+    auto matcher() const
+    {
+        return match::name("gather")(
+            match::args(match::name("reshape")(match::arg(0).bind("data")),
+                        match::is_constant().bind("ind")),
+            match::all_of[match::outputs()](match::used_once(), pointwise()));
+    }
+
+    void apply(module& p, match::matcher_result r) const
+    {
+        auto ins     = r.result;
+        auto ins_rsp = r.instructions["data"];
+        auto ins_ind = r.instructions["ind"];
+        auto ins_pw  = ins->outputs().front();
+
+        // resize input shape
+        auto in_shape = ins_rsp->inputs().front()->get_shape();
+
+        // resize output shape
+        auto out_shape = ins->get_shape();
+
+        // check if output shape is multiple of input shape
+        auto in_lens  = in_shape.lens();
+        auto out_lens = out_shape.lens();
+        if(in_lens.size() != out_lens.size())
+        {
+            return;
+        }
+
+        // if ind is not constant, cannot optimize
+        std::vector<int> vec_ind;
+        auto arg_ind = ins_ind->eval();
+        if(arg_ind.empty())
+        {
+            return;
+        }
+        arg_ind.visit([&](auto v) { vec_ind.assign(v.begin(), v.end()); });
```

Comment:
You need to check the values of `arg_ind`.  It might be easier to recompute the indices for a resize and then check that they match.

Reply:
I have a constant requirement for the ind in the matcher, so I did not put the checker here.

Reply:
anyway, I added the check